MinecraftClient/McClient.cs

```diff
@@ -1084,6 +1085,102 @@ public bool MoveTo(Location location, bool allowUnsafe = false, bool allowDirect
+
+            // If a minimal distance to the goal was given, remove all locations that are too close.
+            if (minDistance>0) 
+            {
+                listOfCloseLocations = listOfCloseLocations.Where(block => block.Distance(goalToApproach) >= minDistance).ToList();
+            }
+
+            // Check how much work to do. Avoid too much effort.
+            if (listOfCloseLocations.Count < 1)
+            {
+                return false;
+            }
+            else if (listOfCloseLocations.Count == 1)
+            {
+                return MoveTo(listOfCloseLocations[0], allowUnsafe);
+            }
+            else
+            {
+                // Create several instances of CalculatePath() to find ways to the closest locations
+                List<Task<Queue<Location>>> listOfPathfindInstances = new List<Task<Queue<Location>>>();
+                CancellationTokenSource cts = new CancellationTokenSource();
+                CancellationToken ct = cts.Token;
+
+                Task pathfindSpawner = Task.Run(() =>
+                {
+                    // Using Semaphore to limit the amount of parallel running tasks
+                    using (SemaphoreSlim concurrencySemaphore = new SemaphoreSlim(pathfindingThreads))
+                    {
+                        foreach (Location closeLoc in listOfCloseLocations)
+                        {
+                            // Wait if there are too many parallel tasks
+                            concurrencySemaphore.Wait();
+
+                            // Create a Task for one path
+                            listOfPathfindInstances.Add(Task<Queue<Location>>.Run(() => Movement.CalculatePath(world, this.location, closeLoc, allowUnsafe)));
+
+                            // If a valid path was found, stop spawning new tasks and exit
+                            if (ct.IsCancellationRequested) { break; }
+                        }
+                    }
+                }, cts.Token);
+
+                // Search through the list of tasks for valid paths
+                // List is sorted from the closest possbile goal to the farthest
+                int i = 0;
+                while(!pathfindSpawner.IsCompleted || i < listOfPathfindInstances.Count)
+                {
+                    if (listOfPathfindInstances.Count > 0)
+                    {
+                        listOfPathfindInstances[i].Wait();
+                        // If there is a valid path
+                        if (listOfPathfindInstances[i].Result != null)
+                        {
+                            // Stop creating new tasks
+                            cts.Cancel();
+                            path = listOfPathfindInstances[i].Result;
+                            return true;
+                        }
+                        else { i++; }
+                    }

```

Comment:
When `pathfindSpawner` is not complete but `i >= listOfPathfindInstances.Count`, this mean the loop must wait for the next task to be added to the list. A tiny Sleep() should save some CPU cycles, right?
```suggestion
                    } else { Thread.Sleep(10); }
```